server/clip_server/model/openclip_model.py

```diff
@@ -20,17 +28,139 @@ class OpenCLIPModel(CLIPModel):
+                    ][-1]
+
+                    def patch_device(module):
+                        try:
+                            graphs = [module.graph] if hasattr(module, "graph") else []
+                        except RuntimeError:
+                            graphs = []
+
+                        if hasattr(module, "forward1"):
+                            graphs.append(module.forward1.graph)
+
+                        for graph in graphs:
+                            for node in graph.findAllNodes("prim::Constant"):
+                                if "value" in node.attributeNames() and str(
+                                    node["value"]
+                                ).startswith("cuda"):
+                                    node.copyAttributes(device_node)
+
+                    model.apply(patch_device)
+                    patch_device(model.encode_image)
+                    patch_device(model.encode_text)
+
+                    # patch dtype to float32 on CPU
+                    if device == "cpu":
+                        float_holder = torch.jit.trace(
+                            lambda: torch.ones([]).float(), example_inputs=[]
+                        )
+                        float_input = list(
+                            float_holder.graph.findNode("aten::to").inputs()
+                        )[1]
+                        float_node = float_input.node()
+
+                        def patch_float(module):
+                            try:
+                                graphs = (
+                                    [module.graph] if hasattr(module, "graph") else []
+                                )
+                            except RuntimeError:
+                                graphs = []
+
+                            if hasattr(module, "forward1"):
+                                graphs.append(module.forward1.graph)
+
+                            for graph in graphs:
+                                for node in graph.findAllNodes("aten::to"):
+                                    inputs = list(node.inputs())
+                                    for i in [
+                                        1,
+                                        2,
+                                    ]:  # dtype can be the second or third argument to aten::to()
+                                        if inputs[i].node()["value"] == 5:
+                                            inputs[i].node().copyAttributes(float_node)
+
+                        model.apply(patch_float)
+                        patch_float(model.encode_image)
+                        patch_float(model.encode_text)
+                        model.float()
+
+                    # ensure image_size attr available at consistent location for both jit and non-jit
+                    model.visual.image_size = model.input_resolution.item()
+                if precision == "fp32":
+                    model = model.float()
```

Comment:
we can simply use `load_openai_model()` here.

Reply:
can't do it because load_openai_model contains openclip's download method which disables us to download model from our s3

Reply:
no, `load_openai_model`  also accept model_path as the parameter.